### PR TITLE
Source Square: Fix Authentication errors

### PR DESCRIPTION
**Pull request by artem1205**
## What
Resolve https://github.com/airbytehq/oncall/issues/1963

## How
Fix spec and manifest

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-square/source_square/spec.yaml [hunk 1]
```diff
@@ -21,9 +21,9 @@ connectionSpecification:
             - client_secret
             - refresh_token
           properties:
-            credentials_title:
+            auth_type:
               type: string
-              const: OAuth Credentials
+              const: Oauth
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-square/source_square/spec.yaml:26]: Let’s re-name to `OAuth`

File: airbyte-integrations/connectors/source-square/source_square/components.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ class AuthenticatorSquare(DeclarativeAuthenticator):
     oauth: DeclarativeOauth2Authenticator
 
     def __new__(cls, bearer, oauth, config, *args, **kwargs):
-        if config.get("api_key"):
+        if config.get("credentials", {}).get("api_key"):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-square/source_square/components.py:26]: What about backward compatibility?

**Review by lazebnyi — APPROVED** [review]
LGTM! Minor comments.

File: airbyte-integrations/connectors/source-square/source_square/components.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-square/source_square/components.py:26]: It was not working; missed spec path

File: airbyte-integrations/connectors/source-square/source_square/spec.yaml (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-square/source_square/spec.yaml]: done

File: airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json [hunk 3]
```diff
@@ -24608,10 +24608,10 @@
               "type": "object",
               "required": [ "auth_type", "api_key" ],
               "properties": {
-                "credentials_title": {
+                "auth_type": {
                   "type": "string",
-                  "const": "API Key",
```
Comment (sherifnada) [inline airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json:24613]: Please keep the original constant as it's more human readable

**Review by sherifnada — CHANGES_REQUESTED** [review]

**Review by sherifnada — APPROVED** [review]
If the constant is fixed I'm happy with merging as long as you tested this in the UI

File: airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json (hunk 3 above)
Comment (artem1205) [reply airbyte-config-oss/init-oss/src/main/resources/seed/oss_catalog.json:24613]: fixed

**Pull request merged**